flytekit/core/array_node.py

```diff
@@ -190,6 +192,16 @@ def execution_mode(self) -> _core_workflow.ArrayNode.ExecutionMode:
         return self._execution_mode
 
     def __call__(self, *args, **kwargs):
+        ctx = FlyteContext.current_context()
+        if ctx.compilation_state is not None:
+            # interface mismatch between the target and the actual inputs since we don't create a new entity with a
+            # transformed to list interface
+            transformed_kwargs = {
```

Comment:
this is fairly hacky. Would love if there's a better way to do this.

Reply:
I think this is fine... but can you pull this out into a separate function and add some comments and that also makes it easier to write a unit test for this.

I assume this is why the bindings in the subnode in the test seem to be binding to the first item.  Could you add a comment somewhere explaining why?

Also does this feature work in local execution?  If so, could you add some more units that just exercise local execution?

Reply:
@wild-endeavor cleaned this up a lot by passing in an "overriden_interface" that is the subnode's interface transformed to a list interface like we do in array node map tasks. By passing in a fake transformed to list interface we can bind all the inputs to the subnode

Reply:
@wild-endeavor and yup this works in local execution. There is a lone unit test for it - let me add more to it.

Reply:
@wild-endeavor there are a couple unit tests to test local execute: https://github.com/flyteorg/flytekit/blob/fb55841f8660b2a31e99381dd06e42f8cd22758e/tests/flytekit/unit/core/test_array_node.py#L75 + https://github.com/flyteorg/flytekit/blob/fb55841f8660b2a31e99381dd06e42f8cd22758e/tests/flytekit/unit/core/test_array_node.py#L99